h2o-algos/src/main/java/hex/DataInfo.java

```diff
@@ -973,7 +973,6 @@ public final int getCategoricalId(int cid, int val) {
     int [] offs = fullCatOffsets();
     int expandedVal = val + offs[cid];
     if(expandedVal >= offs[cid+1]) {  // previously unseen level
-      assert _valid:"Categorical value out of bounds, got " + val + ", next cat starts at " + fullCatOffsets()[cid+1];
```

Comment:
Looks like this address the issue with the validation set containing enum values that are not found in training data set.  It just throw an error.  Just talked to Megan K, usually, for test dataset, it will replace the unseen enum value with the mode.  By commenting out this assertion, it should be okay.